MIP/mip-n/README.md

```diff
@@ -0,0 +1,148 @@
+
+```
+<software-unit>-<environment>
+```
+
+An example of how this can be released in Git and GitHub is provided in the following section.
+
+### Standardized Source Control
+1. Source control MUST be responsible for initiating all releases with the following tags:
+    1. A tag corresponding to the Standardized Release Naming Convention above.
+    2. A tag corresponding to the commit hash of the release.
+    3. A tag corresponding to the environment indirection for the release should it be valid.
+
+2. Releases corresponding to environment indirection MUST only be tagged by making a release from the default branch of the repository.
+
+3. By default, the `<release-type>` SHOULD correspond to the branch from which the release is made.
+
+4. Tag indirection CAN be performed as follows:
+
+```bash
+git tag v1.0.0
+git push origin v1.0.0
+git tag -f latest v1.0.0 
+git push origin -f latest 
+```
+
+5. GitHub SHOULD be the default source control platform. Within the platform each release should contain at least the archive of the source code which would produce the release artifact.
+
+6. Well-known configuration files MUST also be included in the release archive where they apply. 
+
+7. Releases outside of GitHub, e.g., DockerHub, MUST be tagged with the same tags as the GitHub release and linked in the release notes on the GitHub release.
+
+8. Releases to established environments MUST follow governance guidelines for the software unit in question.
+
+9. Releases to established environments MUST trigger an update of all usage of that software unit to the appropriate tag by Movement Labs. This SHOULD be automated to the extent possible to ensure that the source control reflects the canonical release.
+
+### Programmatic Access to Release Artifacts
+1. Source control SHOULD be the primary source of release artifacts. 
+
+2. Artifacts which are in consumable forms such as container images, binaries, or libraries can be consumed as the developer sees fit.
+
+3. We recommend the following Rust pseudocode API for fetching artifacts such as well-known configuration files, contract addresses, or other artifacts which may need an application context to be properly consumed:
+
+```rust
+trait Release<T> {
+    fn fetch_artifact(&self) -> Result<T, Error>;
+}
+
+/// Downloads a well-known file from the release in the source control platform.
+struct WellKnownFileRelease {
+    source: String,
+    destination: String
+}
+
+/// The downloaded well-known file.
+struct WellKnownFileArtifact {
+    path: String,
+}
+
+impl Release<WellKnownFileArtifact> for WellKnownFileRelease {
+    fn fetch_artifact(&self) -> Result<WellKnownFileArtifact, Error> {
+       /// ...
+    }
+}
+
+/// Gets the value of a JSON field from a release file.
+struct JSONFieldRelease {
+    source: String,
+    field: String
+}
+
+/// The value of the JSON field.
+struct JSONFieldArtifact {
+    value: Value
+}
```

Comment:
This looks like unnecessary overengineering, a simple intrinsic method on the metadata handle could be enough:

```rust
impl ReleaseMetadata {
    pub fn fetch_field(&self, field_name: impl AsRef<str>) -> Result<Value, Error>
}
```

Reply:
Also, if there is a desire to make the API more strongly typed than a JSON value bag, there should perhaps be more strongly typed methods for specific fields, e.g.

```rust
impl ReleaseMetadata {
    // Every successfully fetched release metadata must have a valid version,
    // so this is infallible:
    pub fn version(&self) -> semver::Version {
        // ...
    }

    pub fn artifact_uri(&self, kind: ArtifactKind) -> Option<Uri> {
        // ...
    }
}
```

Reply:
I also think the more specific API illustrates the intent more clearly.
As a developer, I don't know if the release metadata should be JSON, but I'd be interested in some specific data fields that this specification agrees should be provided.